frappe/core/doctype/document_comparator/document_comparator.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2023, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+import json
+
+import frappe
+from frappe import _
+from frappe.core.doctype.version.version import get_diff
+from frappe.model.document import Document
+
+
+class DocumentComparator(Document):
+	# begin: auto-generated types
+	# This code is auto-generated. Do not modify anything in this block.
+
+	from typing import TYPE_CHECKING
+
+	if TYPE_CHECKING:
+		from frappe.types import DF
+
+		doctype_name: DF.Link
+		document: DF.DynamicLink
+	# end: auto-generated types
+	pass
+
+	def validate(self):
+		self.validate_doctype_name()
+		self.validate_document()
+
+	def validate_doctype_name(self):
+		if not self.doctype_name:
+			frappe.throw(_("{} field cannot be empty.").format(frappe.bold("Doctype")))
+
+	def validate_document(self):
+		if not self.document:
+			frappe.throw(_("{} field cannot be empty.").format(frappe.bold("Document")))
+
+	@frappe.whitelist()
+	def compare_document(self):
+		self.validate()
+		amended_document_names = frappe.db.get_list(
```

Comment:
Here the user has to select the first cancelled doc for any results to show up, ideally, the user will select the latest submitted doc to check the differences, here instead of getting the values using like, last 5 amendments can be fetched from the amend_from field.